.github/workflows/tf-aws-test.yml

```diff
@@ -5,39 +5,63 @@ on:
       - published
   workflow_dispatch: {}
 
+env:
+  AWS_REGION : "us-east-1"
+
 jobs:
-  install:
-    name: Installation & Test
-    strategy:
-      fail-fast: true
+  setup:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v3
+    - name: Checkout Repository
+      uses: actions/checkout@v3
+    - name: Get list of directories
+      id: setdirs
+      shell: bash
+      run: |
+        dirs=$(ls -d examples/tests/sdk_tests/*/ | sed 's/\/$//' | grep -v "external" | jq -R -s -c 'split("\n")[:-1]')
+        processed_dirs=$(echo "{ \"directory\": $dirs }" | jq -c '[ .directory[] | {directory: ., name: (split("/") | last)}]')
+        wrapped_dirs=$(echo "{ \"test\": $processed_dirs }" | jq -c .)
+        echo "DIRS=$wrapped_dirs" >> $GITHUB_ENV
```

Comment:
Perhaps add a small script in the `scripts/` folder to handle this logic?

Reply:
I'm not sure this script, as is, is reusable- as it reads the directories from `examples/tests/sdk_tests/*/` and saves the output in some sort of object as an env nar, I don't think we use it again or need to save an env var when running locally...
If we want to enable this workflow to run locally, we'll have to rewrite the whole thing (considering we're not running on 14 different machines) or just use [act](https://github.com/nektos/act),

Reply:
I agree! On the other hand, chatGPT did a pretty good job at explaining that, the next person to maintain it will be just fine